NSS-Day-Cohort-53/tabloid-full-stack-always-picturesque-icebergs
# NC-added the create a new tag functionality

#### Changes Made

1. Modified file `ITagRepository.cs` to include `Tag getTagbyId(Int id), and void Add(Tag tag)` and functionality.
2. Modified file `TagRepository.cs` to include `public void add(Int id),  and public Tag getTagById(int id)` and functionality.
3. Modified file `TagController.cs` to include `IActionResult Get(int id), and public IActionResult Post(Tag tag) ` and functionality.
 1. Added file `TagCreate.js` to `Componets` directory.
 

#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout nc-addTag
    ```
4. Open a new Terminal tab (⌘T) and navigate to the server directory.
5. Test app functionality.
    When first running the code. Head to Tag management once there click on a link that says create a tag. Then you will be sent to a new page that say create a tag. Once there you type in a new tag and click the submit button. It will send you back to the Tag management page with a new tag on the list.
6. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.
Changes made after review by Fred
1.got rid of the white space
2. deleted the committed code
3. removed the unused items from the TagCreate.js
Changes made after being reviewed by Josh
1. On TagControler.cs made changes  to Tag getTagById(int id) adding PascalCasing 
2. On TagRepository.cs changed  added DbUtils to replace the cmd on the TagRepository.cs
3. On TagRepository.cs added PascalCasing to the GetTagById
4. ApplicationViews.js made changes to make it consistent with the naming conventions with teams' category route path by changing the path to a lower case tag and adding exact to the end of line 39. Also changed Line 47 to /tag/new to make everything match.
5. TagCreate.js  Added an if statement inside the handelSaveFunction that looks for an empty string and trim() to stop a user from entering in spaces.
https://github.com/NSS-Day-Cohort-53/tabloid-full-stack-always-picturesque-icebergs/projects/1#card-80770758 closed
